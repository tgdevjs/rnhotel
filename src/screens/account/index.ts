export * from './Account';
export * from './AccountGuest';
export * from './AccountMenu';
export * from './JoinModal';
export * from './SignInModal';
