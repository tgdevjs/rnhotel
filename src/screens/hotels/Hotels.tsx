import React, { Component } from "react";
import { ImageBackground, FlatList, StyleSheet, View } from "react-native";
import { NavigationScreenProp } from "react-navigation";
import Icons from "react-native-vector-icons/MaterialCommunityIcons";
import LinearGradient from "react-native-linear-gradient";
import { TouchableHighlight } from "react-native-gesture-handler";

import { HotelTitle } from "../../components";

type Props = {
  navigation: NavigationScreenProp<any, any>;
};

const styles = StyleSheet.create({
  iconRow: {
    flexDirection: "row"
  },
  item: {
    justifyContent: "flex-end",
    height: 160,
    width: "100%"
  },
  itemTitleBar: {
    marginHorizontal: 8,
    marginVertical: 8
  },
  icons: {
    margin: 2
  },
  itemSeparator: {
    backgroundColor: "white",
    height: 10
  }
});

const data = Array.from({ length: 30 }, (v, i) => ({
  key: String(i),
  name: `hotel-${i}`,
  imageSrc: "http://lorempixel.com/320/280/city",
  price: "$120"
}));

const renderIconRow = () => {
  const icons = [
    "key-variant",
    "bus-side",
    "car",
    "wifi",
    "smoking-off",
    "silverware-fork-knife"
  ].map(v => (
    <Icons
      style={styles.icons}
      key={`${v}`}
      name={v}
      size={25}
      fontWeight="800"
      color="white"
    />
  ));

  return <View style={styles.iconRow}>{icons}</View>;
};

export const Hotels = ({ navigation: { navigate } }: Props) => {
  return (
    <FlatList
      data={data}
      ItemSeparatorComponent={() => <View style={styles.itemSeparator} />}
      renderItem={({ item, item: { name, imageSrc } }) => (
        <TouchableHighlight
          onPress={() => navigate("ConfirmReservation", { hotel: item })}
        >
          <ImageBackground source={{ uri: imageSrc }} style={styles.item}>
            {/* <View style={styles.item}> */}
            <LinearGradient colors={["#192f6a00", "#3b599888", "#4c669fff"]}>
              <View style={styles.itemTitleBar}>
                <HotelTitle title={name} />
                {renderIconRow()}
              </View>
            </LinearGradient>
            {/* </View> */}
          </ImageBackground>
        </TouchableHighlight>
      )}
    />
  );
};
