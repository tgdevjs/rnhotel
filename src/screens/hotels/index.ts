export * from './Hotels';
export * from './ConfirmReservation';
