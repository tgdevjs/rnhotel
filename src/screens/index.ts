export * from './account';
export * from './hotels';
export * from './search';
export * from './stays';
