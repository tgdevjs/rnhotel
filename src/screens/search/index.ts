export * from './Search';
export * from './SearchCalendar';
