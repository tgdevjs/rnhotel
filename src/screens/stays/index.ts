export * from './Stays';
export * from './StayItem';
export * from './StaysList';
export * from './StaysListGuest';
