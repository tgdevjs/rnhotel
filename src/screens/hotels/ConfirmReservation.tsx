import React from 'react';
import { Button, StyleSheet, Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { StackActions, NavigationActions } from 'react-navigation';
import { compose, graphql } from 'react-apollo';
import LinearGradient from 'react-native-linear-gradient';
import cuid from 'cuid';

import { DateDisplay, HotelTitle, SignInButton } from '../../components';
import { withSearchMutation, withSearchQuery } from '../../apollo/client-state/search';
import { withUserQuery } from '../../apollo/client-state/user';
import { reservationsQuery, createReservationMutation } from '../../apollo/queries';
import { getReservationsQueryVariables } from '../../utils';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-between',
    flex: 1,
  },
  header: {
    alignItems: 'center',
    justifyContent: 'center',
    height: 200,
  },
  dateDisplay: {
    paddingBottom: 30,
  },
  hotelTitle: {
    color: 'black',
  },
  main: {
    flex: 1,
    justifyContent: 'center',
  },
  buttonSection: {
    marginBottom: 40,
    justifyContent: 'center',
    alignItems: 'center',
  },
  actionSection: {
    margin: 20,
    alignItems: 'center',
  },
  actionText: {
    fontSize: 16,
  },
  labelText: {
    fontSize: 16,
  },
  valueText: {
    fontSize: 20,
    fontWeight: '600',
  },
});

export class ConfirmReservation extends React.Component {
  state = { hasCreatedReservation: false };

  id = cuid();

  renderContent() {
    const {
      createReservation,
      endDay,
      navigation: { dispatch, getParam, navigate, popToTop },
      startDay,
      user: { name },
    } = this.props;
    const { name: hotelName } = getParam('hotel');

    const variables = {
      data: {
        id: this.id,
        name,
        hotelName,
        arrivalDate: startDay,
        departureDate: endDay,
      },
    };
    const { hasCreatedReservation } = this.state;

    // User not logged in
    if (!name) {
      return (
        <View style={styles.buttonSection}>
          <View style={styles.actionSection}>
            <Text style={styles.labelText}>Sign in to continue.</Text>
          </View>
          <SignInButton onPress={() => navigate('SignIn')} />
        </View>
      );
    }

    if (!hasCreatedReservation) {
      return (
        <View style={styles.buttonSection}>
          <View style={styles.actionSection}>
            <Text style={styles.labelText}>User Name:</Text>
            <Text style={styles.valueText}>{name}</Text>
          </View>
          <Button
            title="Book Now"
            onPress={async () => {
              await createReservation({
                variables,
                refetchQueries: [
                  { query: reservationsQuery, variables: getReservationsQueryVariables({ name }) },
                ],
              });
              this.setState({ hasCreatedReservation: true });
            }}
          />
        </View>
      );
    }

    return (
      <View style={styles.buttonSection}>
        <View style={styles.actionSection}>
          <Text style={styles.valueText}>Success!</Text>
          <Text style={styles.labelText}>Your reservation was created.</Text>
        </View>
        <Button title="Start New Search" onPress={() => popToTop()} />
        <Button
          title="View Reservations"
          onPress={() => {
            // Reset Main stack
            dispatch(
              StackActions.reset({
                index: 0,
                actions: [NavigationActions.navigate({ routeName: 'Main' })],
              })
            );
            navigate('Stays');
          }}
        />
      </View>
    );
  }

  render() {
    const {
      endDay,
      navigation: { getParam },
      startDay,
    } = this.props;
    const { name: hotelName } = getParam('hotel');

    return (
      <View style={styles.container}>
        <LinearGradient colors={['#192f6a00', '#3b599888']} style={styles.header}>
          <DateDisplay style={styles.dateDisplay} startDay={startDay} endDay={endDay} />
          <HotelTitle style={styles.hotelTitle} title={hotelName} />
          <Text>{this.id}</Text>
        </LinearGradient>
        <View style={styles.main}>{this.renderContent()}</View>
      </View>
    );
  }
}

export const ConfirmReservationQuery = compose(
  withSearchQuery,
  withSearchMutation,
  withUserQuery,
  graphql(createReservationMutation, { name: 'createReservation' })
)(ConfirmReservation);

ConfirmReservation.propTypes = {
  createReservation: PropTypes.func.isRequired,
  endDay: PropTypes.string.isRequired,
  navigation: PropTypes.shape({}).isRequired,
  startDay: PropTypes.string.isRequired,
  user: PropTypes.shape({
    name: PropTypes.string,
  }).isRequired,
};
