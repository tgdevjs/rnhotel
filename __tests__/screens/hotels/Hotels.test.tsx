import React from "react";
import { Hotels } from "../../../src/screens/hotels";

function hotels(props = {}) {
  const defaultProps = {
    navigation: { navigate: jest.fn() },
    ...props
  };
  return shallow(<Hotels {...defaultProps} />);
}

describe("Hotels", () => {
  it("renders default", () => {
    expect(hotels()).toMatchSnapshot();
  });
});
