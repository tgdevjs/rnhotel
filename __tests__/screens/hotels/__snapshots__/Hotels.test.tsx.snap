// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotels renders default 1`] = `
<FlatList
  ItemSeparatorComponent={[Function]}
  data={
    Array [
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "0",
        "name": "hotel-0",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "1",
        "name": "hotel-1",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "2",
        "name": "hotel-2",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "3",
        "name": "hotel-3",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "4",
        "name": "hotel-4",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "5",
        "name": "hotel-5",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "6",
        "name": "hotel-6",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "7",
        "name": "hotel-7",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "8",
        "name": "hotel-8",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "9",
        "name": "hotel-9",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "10",
        "name": "hotel-10",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "11",
        "name": "hotel-11",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "12",
        "name": "hotel-12",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "13",
        "name": "hotel-13",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "14",
        "name": "hotel-14",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "15",
        "name": "hotel-15",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "16",
        "name": "hotel-16",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "17",
        "name": "hotel-17",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "18",
        "name": "hotel-18",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "19",
        "name": "hotel-19",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "20",
        "name": "hotel-20",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "21",
        "name": "hotel-21",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "22",
        "name": "hotel-22",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "23",
        "name": "hotel-23",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "24",
        "name": "hotel-24",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "25",
        "name": "hotel-25",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "26",
        "name": "hotel-26",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "27",
        "name": "hotel-27",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "28",
        "name": "hotel-28",
        "price": "$120",
      },
      Object {
        "imageSrc": "http://lorempixel.com/320/280/city",
        "key": "29",
        "name": "hotel-29",
        "price": "$120",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
